compiler/qsc_rca/tests/vars.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+pub mod common;
+
+use common::{check_last_statement_compute_properties, CompilationContext};
+use expect_test::expect;
+
+#[test]
+fn check_rca_for_callable_var() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(r#"H"#);
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Classical
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_udt_var() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+            open Microsoft.Quantum.Math;
+            Complex"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Classical
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_static_int_var() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+            let a = 42;
+            a"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Classical
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_static_qubit_var() {
+    let mut compilation_context = CompilationContext::new();
+    compilation_context.update(
+        r#"
+            use q = Qubit();
+            q"#,
+    );
+    let package_store_compute_properties = compilation_context.get_compute_properties();
+    check_last_statement_compute_properties(
+        package_store_compute_properties,
+        &expect![
+            r#"
+            ApplicationsGeneratorSet:
+                inherent: Quantum: QuantumProperties:
+                    runtime_features: RuntimeFeatureFlags(0x0)
+                    value_kind: Element(Static)
+                dynamic_param_applications: <empty>"#
+        ],
+    );
+}
+
+#[test]
+fn check_rca_for_dynamic_result_var() {
```

Comment:
this might be getting too complex, but I'm curious what the analysis would say for things like locally defined lambdas that capture a value. That opens the door to the lambda itself being dynamic, the lambda being static but the captured variables being dynamic, etc. Might be out of scope for now though.

Reply:
The analysis for lambdas is currently not thorough at all. We basically just mark it with the `UseOfClosure` runtime feature and consider any lambda as dynamic. This has the consequence that the output of calling a lambda is also dynamic.

I added some tests to document the current expectations from lambdas both in (new) closures.rs and calls.rs test files.